src/gui/options_banlist.cpp

```diff
@@ -62,16 +72,29 @@ void BanListOptions::on_buttonBox_accepted()
     accept();
 }
 
-void BanListOptions::on_pb_BanIP_clicked()
+bool BanListOptions::banIP(const QString &ip)
 {
-    QString s = m_ui->ed_IP->text();
-    if (s.isEmpty()) {
+    if (!isValidIP(ip)) {
         QMessageBox::warning(this, tr("Warning"), tr("Entered IP address is invalid."));
-        return;
+        return false;
+    }
+    QHostAddress addr(ip);
```

Comment:
`addr` is not redundant variable. For example  `QHostAddres("1.2").toString()` returns "1.0.0.2" Also "0:0:0:0:0::1" -> "::1"
E.g. there is probability of change of IP string to normalized state using such trick.